### PR TITLE
DEVELOPER-3009 - Register site users using social providers.

### PR DESCRIPTION
**Pull request by RubyGemTSL**

### CODE REVIEW

File: _cucumber/features/login_register/social_registration.feature [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  Scenario: 3 - Registration using GitHub login which contains all mandatory information (first name, last name, but email already exists). User changes email to create new account.
+    Given I am on the Registration page
+    When I try to link a GitHub account to an existing account
+    Then I should see a warning that the email is already registered
+     And I select to Choose another email
+     And I complete the required additional information with a new email address
+    Then I should be registered and logged in
+
+  @rhd_social_test_user01 @logout @github_teardown
+  Scenario: 4 - Registration using GitHub login which doesn't contain some mandatory information (first name, last name), email is unique. User is asked to fill in mandatory info during login.
```
Comment (velias) [inline _cucumber/features/login_register/social_registration.feature]: Is this scenario finished? I miss some steps there like filling in data and verifying user is registered and logged in.

File: _cucumber/features/login_register/social_registration.feature [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    Given I am on the Registration page
+    When I register a new account using a GitHub account that contains missing profile information
+    Then I should be asked to fill in mandatory information with a message "In creating your Red Hat Developers user profile, it looks like we were not able to get some required information. Fill in the following fields to finalize your user profile please."
+
+  @rhd_social_test_user01 @keycloak_social_teardown @logout @github_teardown
+  Scenario: 5 - Registration using GitHub login which doesn't contain some mandatory the information (first name, last name), email already exists. User links social provider to existing account
+    Given I am on the Registration page
+    When I try to register using a GitHub account that contains missing profile information$ with an existing RHD registered email
```
Comment (velias) [inline _cucumber/features/login_register/social_registration.feature]: is $ in `information$` on this line intentional?

File: _cucumber/features/login_register/social_registration.feature [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    Then I should see a warning that the email is already registered
+      And I click on the Send Verification email button
+    Then I should receive an email containing a verify email link
+      And I navigate to the verify email link
+    Then I should be registered and logged in
+
+  @rhd_social_test_user02 @keycloak_social_teardown @logout @github_teardown
+  Scenario: 6 - Registration using GitHub login which doesn't contain some mandatory information (first name, last name), email that already exists. User changes email to create new account.
+    Given I am on the Registration page
+    When I try to register using a GitHub account that contains missing profile information$ with an existing RHD registered email
```
Comment (velias) [inline _cucumber/features/login_register/social_registration.feature]: is $ in `information$` on this line intentional?

File: _cucumber/features/login_register/social_registration.feature (hunk 1 above)
Comment (RubyGemTSL) [inline _cucumber/features/login_register/social_registration.feature]: Will add missing steps. thanks!

File: _cucumber/features/login_register/social_registration.feature (hunk 2 above)
Comment (RubyGemTSL) [inline _cucumber/features/login_register/social_registration.feature]: Nope, good spot, i'll remove.

File: _cucumber/features/login_register/social_registration.feature (hunk 3 above)
Comment (RubyGemTSL) [inline _cucumber/features/login_register/social_registration.feature]: Nope, good spot, i'll remove.

**Comment by RubyGemTSL** [comment]
there is two failing tests on this. We cannot seem to access the gmail API on stumpjumper in order to get verification emails. I've marked the IP as a known safe IP but still the issue persists. Will continue to investigate.

**Comment by RubyGemTSL** [comment]
retest this please

**Comment by RubyGemTSL** [comment]
retest this please

**Comment by RubyGemTSL** [comment]
retest this please

**Comment by RubyGemTSL** [comment]
retest this please

**Comment by RubyGemTSL** [comment]
retest this please

**Comment by RubyGemTSL** [comment]
retest this please

**Comment by RubyGemTSL** [comment]
@paulrobinson - this is now ready to be merged (when Blinkr is finished) as @velias is happy with the scenarios. cc @LightGuard

**Comment by RubyGemTSL** [comment]
retest this please

**Comment by LightGuard** [comment]
Blinkr is taking a long time. I'd suggest not waiting for it currently.

**Comment by RubyGemTSL** [comment]
@LightGuard this is good to go now if you're OK to merge.

**Pull request merged**